tests/rptest/tests/consumer_group_balancing_test.py

```diff
@@ -0,0 +1,201 @@
+                # a sigle replica per __consumer_offsets partition.
+                # this will change with implementation of
+                # even distribution control of __consumer_offsets leaders
+                "internal_topic_replication_factor": 1,
+                # a __consumer_offsets partiton per cgroup
+                "group_topic_partitions": cgroup_balancing_test_num_groups,
+            },
+            **kwargs)
+
+    def make_consumer_properties(base_properties, instance_id=None):
+        properties = {}
+        properties.update(base_properties)
+        if instance_id:
+            properties['group.instance.id'] = instance_id
+        return properties
+
+    def create_consumer(self,
+                        topic,
+                        group,
+                        instance_id=None,
+                        consumer_properties={}):
+        return KafkaCliConsumer(self.test_context,
+                                self.redpanda,
+                                topic=topic,
+                                group=group,
+                                from_beginning=True,
+                                formatter_properties={
+                                    'print.value': 'false',
+                                    'print.key': 'false',
+                                    'print.partition': 'true',
+                                    'print.offset': 'true',
+                                },
+                                consumer_properties=ConsumerGroupBalancingTest.
+                                make_consumer_properties(
+                                    consumer_properties, instance_id))
+
+    def consumed_at_least(consumers, count):
+        return all([len(c._messages) > count for c in consumers])
+
+    def validate_group_state(self,
+                             group,
+                             expected_state,
+                             static_members,
+                             group_members: int = 2):
+        rpk = RpkTool(self.redpanda)
+        # validate group state
+        rpk_group = rpk.group_describe(group)
+
+        assert rpk_group.members == group_members
+        assert rpk_group.state == expected_state
+
+        for p in rpk_group.partitions:
+            if static_members:
+                assert 'panda-consumer' in p.instance_id
+            else:
+                assert p.instance_id is None
+
+    def setup_producer(self, p_cnt):
+        # create topic
+        self.topic_spec = TopicSpec(partition_count=p_cnt,
+                                    replication_factor=3)
+        self.client().create_topic(specs=self.topic_spec)
+        # produce some messages to the topic
+        self.producer = RpkProducer(self._ctx, self.redpanda,
+                                    self.topic_spec.name, 128, 5000, -1)
+        self.producer.start()
+
+    @cluster(num_nodes=cgroup_balancing_test_num_brokers +
+             cgroup_balancing_test_num_groups + 1)
```

Comment:
While KafCliConsumer expects one node per consumer, it isn't really necessary.  
KgoVerifierConsumerGroupConsumer for example will run several consumers in one instance of the service.

The fewer nodes the test uses, the more it is leaving for other tests to run concurrently -> faster overall execution.

Reply:
Good point, I wasn't aware of other consumers with different node assignment properties. Is the `KgoVerifierConsumerGroupConsumer` the one you think would work in this case if I wanted to save on consumer nodes? Please advise

Reply:
If you want to avoid modifying any services, you can use KgoVerifierConsumerGroupConsumer and KgoVerifierProducer (the producer writes the data patterns that the consumer expects), and then inherit from PreallocNodesTest to get one extra node that you can pass into that service.

There's no good reason that those are the only services that support passing in explicit nodes, it just that the others are all written to own an entire node by default.  If you're interested, I think it would be quite legitimate to modify another simpler consumer service to take an optional `node` argument to re-use another node rather than allocating one.  See `KgoVerifierService` for an example of how to do it: you basically just pass num_node=0 to the Service constructor and then assign self.nodes yourself, and override `free` to avoid freeing the node.